test/db/index.js

```diff
@@ -0,0 +1,449 @@
+      contractAddress: null,
+      cumulativeGasUsed: 207559,
+      gasUsed: 207559,
+      gasPrice: '30000000',
+    };
+    describe('insert', () => {
+      it('should start with empty transactionReceipts db', async () => {
+        txReceiptsCount = await db.TransactionReceipts.count({});
+        txReceiptsCount.should.equal(0);
+      });
+
+      it('should insert with new transaction receipt', async () => {
+        await DBHelper.insertTransactionReceipt(toBeInsertedTxReceipt);
+        txReceipt = await db.TransactionReceipts.find({});
+        txReceipt.length.should.equal(1);
+        expect(txReceipt[0]).excluding("_id").to.deep.equal(toBeInsertedTxReceipt)
+      });
+
+      it('should insert with existed transaction Hash', async () => {
+        await DBHelper.insertTransactionReceipt(toBeInsertedExistedTxReceipt);
+        txReceipt = await db.TransactionReceipts.find({});
+        txReceipt.length.should.equal(1);
+        expect(txReceipt[0]).excluding(["_id", "gasPrice"]).to.deep.equal(toBeInsertedExistedTxReceipt)
+        txReceipt[0].gasPrice.should.equal(toBeInsertedTxReceipt.gasPrice);
+      });
+    });
+
+    describe('find one transaction receipt', () => {
+      it('should find existed transaction Hash', async () => {
+        txReceipt = await DBHelper.findOneTransactionReceipt({transactionHash: toBeInsertedExistedTxReceipt.transactionHash});
+        should.exist(txReceipt);
+        expect(txReceipt).excluding(["_id", "gasPrice"]).to.deep.equal(toBeInsertedExistedTxReceipt)
+        txReceipt.gasPrice.should.equal(toBeInsertedTxReceipt.gasPrice);
+      })
+
+      it('should not find non-existed transaction Hash', async () => {
+        txReceipt = await DBHelper.findOneTransactionReceipt({transactionHash: 123321});
+        should.not.exist(txReceipt)
+      })
+    })
+
+    describe('update transaction receipt', () => {
+      it('should update existed transaction Hash transaction receipt', async () => {
+        const toBeUpdatedExistedTxReceipt = {
+          status: true,
+          blockHash: '0x67b323ds35531840561747a4c1bf9aa32a31641367ae1270db74cc321c9eb78e82f',
+          blockNum: 143234534,
+          transactionHash: '0x81fcd2aec9f2f45a96eea6deb858a76fecce619df091f62ec92ec3bce6181d18',
+          from: '0x47ba776b3ed5230ffa7e',
+          to: '0x809388c8770c578cb873o2399436d8156fda',
+          contractAddress: '0x47ba776b3ed5230ffa7e',
+          cumulativeGasUsed: 234359,
+          gasUsed: 324432,
+          gasPrice: '60000000',
+        };
+        await DBHelper.updateTransactionReceipt(toBeUpdatedExistedTxReceipt);
+        txReceipt = await DBHelper.findOneTransactionReceipt({transactionHash: toBeUpdatedExistedTxReceipt.transactionHash});
+        expect(txReceipt).excluding("_id").to.deep.equal(toBeUpdatedExistedTxReceipt)
+      })
+    })
+
+    it('should fail update non-existed transaction Hash transaction receipt', async () => {
```

Comment:
should throw trying to update a non-existing tx receipt

Reply:
it doesn't throw error. so maybe leave it as should fail?

Reply:
gotcha, fail is fine